### PR TITLE
fix(insights): fix stacked line chart

### PR DESCRIPTION
**Pull request by aspicer**
## Problem

As reported here:
https://github.com/PostHog/posthog/issues/22596

Line charts are displaying as stacked

https://github.com/PostHog/posthog/assets/1855120/385c81a3-e429-4127-9d1e-656f8b43c66c

## Changes

https://github.com/PostHog/posthog/assets/1855120/cbafef3d-2bf1-49b8-b4eb-eb16a075919c

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Ran it locally

### CODE REVIEW

File: frontend/src/scenes/insights/insightVizDataLogic.ts [hunk 1]
```diff
@@ -138,12 +135,7 @@ export const insightVizDataLogic = kea<insightVizDataLogicType>([
                 display !== ChartDisplayType.WorldMap &&
                 dateRange?.date_from !== 'all',
         ],
-        supportsPercentStackView: [
-            (s) => [s.querySource, s.display],
-            (q, display) =>
-                isTrendsQuery(q) &&
-                PERCENT_STACK_VIEW_DISPLAY_TYPE.includes(display || ChartDisplayType.ActionsLineGraph),
-        ],
+        supportsPercentStackView: [(s) => [s.querySource, s.display], supportsPercentStackView],
```
Comment (aspicer) [inline frontend/src/scenes/insights/insightVizDataLogic.ts:138]: image

@Twixes Is there any reason why these simple passthrough util functions need to be written with args like all the ones in this picture?

File: frontend/src/queries/nodes/InsightViz/utils.ts [hunk 2]
```diff
@@ -113,8 +114,11 @@ export const getShowLabelsOnSeries = (query: InsightQueryNode): boolean | undefi
     return undefined
 }
 
+export const supportsPercentStackView = (q: InsightQueryNode, display?: ChartDisplayType): boolean =>
+    isTrendsQuery(q) && PERCENT_STACK_VIEW_DISPLAY_TYPE.includes(display || ChartDisplayType.ActionsLineGraph)
```
Comment (Twixes) [inline frontend/src/queries/nodes/InsightViz/utils.ts]: Looks like we don't actually need the `display` arg?

File: frontend/src/queries/nodes/InsightViz/utils.ts [hunk 3]
```diff
@@ -113,8 +114,11 @@ export const getShowLabelsOnSeries = (query: InsightQueryNode): boolean | undefi
...
+
 export const getShowPercentStackView = (query: InsightQueryNode): boolean | undefined => {
-    if (isTrendsQuery(query)) {
+    if (isTrendsQuery(query) && supportsPercentStackView(query, query.trendsFilter?.display)) {
```
Comment (Twixes) [inline frontend/src/queries/nodes/InsightViz/utils.ts:123]: `supportsPercentStackView` already performs `isTrendsQuery`

**Review by Twixes — COMMENTED** [review]
Sounds good to me, maybe can be simplified a bit

File: frontend/src/scenes/insights/insightVizDataLogic.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/scenes/insights/insightVizDataLogic.ts]: I don't see one, but then JS is full of surprises

**Review by Twixes — APPROVED** [review]

**Pull request merged**